modules/seqexec/engine/src/main/scala/seqexec/engine/Sequence.scala

```diff
@@ -231,12 +237,27 @@ object Sequence {
     }
 
     /**
-      * Initialize a `State` passing a `Queue` of pending `Sequence`s.
+      * Initialize a `State` passing a `Sequence` of pending `Step`s.
       */
     // TODO: Make this function `apply`?
     def init(q: Sequence): State =
       Sequence.Zipper.zipper(q).map(Zipper(_, SequenceState.Idle))
-        .getOrElse(Final(Sequence.empty(q.id), SequenceState.Idle))
+        .getOrElse(Final(q, SequenceState.Idle))
+
+    /**
+      * Rebuilds the state of a sequence with a new steps definition, but preserving breakpoints and skip marks
+      * The sequence must not be running.
+      * @param q New sequence definition
+      * @param st Old sequence state
+      * @return The new sequence state
+      */
+    def reload(steps: List[Step], st:State): State =
```

Comment:
Important: `reload` uses the step state from `steps`. In the case of reloading from the ODB, this means that if a step was run with the old Seqexec, that fact will be reflected in the sequence state.

Reply:
Is this called on `Sync`?

Reply:
Yes